### PR TITLE
Update Zola to 0.18.0

### PR DESCRIPTION
**Pull request by TrialDragon**
Updates Zola to 0.18.0.

Adopted #719 
Unblocks #710 , probably

### CODE REVIEW

**Comment by TrialDragon** [comment]
Okay, cool. So the issue we were having with updating Zola was https://github.com/shalzz/zola-deploy-action/issues/71 and https://github.com/shalzz/zola-deploy-action/issues/78 : an issue with zola-deploy-action where the proper solution prior to `v0.18.0` would've been to use the hotfix version `v0.17.2-1` per https://github.com/shalzz/zola-deploy-action/issues/81#issuecomment-1837382452 .

**Comment by TrialDragon** [comment]
Thanks @BD103 for mentioning the slash as division deprecation / removal in Sass.

**Comment by TrialDragon** [comment]
To address other updating concerns:
* The new `authors` content variable seems to be just for feeds and isn't accessible in templates; so we can't use this first class variable in place of our current `extra.author` variable.
* Zola's Built-in Shortcodes removal; we luckily don't use them.
* `index.md` and `_index.md` can't be in same folder; luckily we don't have that.
* Files that get process—primarily images—now incorporate the stem; we do use resize_image which this affects, but this shouldn't break anything as far as I'm aware.

**Comment by TrialDragon** [comment]
Manually ran and checked: Examples, Assets, Errors, and Community (people). They render correctly and are not broken.
Per https://discord.com/channels/691052431525675048/695741366520512563/1208861386097172500 addressing concerns mentioned on Discord.

**Review by doup — CHANGES_REQUESTED** [review]
I'm pretty sure using SASS `math.div` inside CSS `calc()` without SASS interpolation is a bug. Actually, it's unnecessary, as the browser supports division. In those cases you can just leave it as it was.

File: sass/components/_asset-card.scss [hunk 1]
```diff
@@ -14,7 +15,7 @@ $asset-card-padding: 0.4rem;
     grid-area: banner;
     display: block;
     position: relative;
-    aspect-ratio: 16/9;
+    aspect-ratio: math.div(16, 9);
```
[the remark this answers is not in the document]
Comment (TrialDragon) [reply sass/components/_asset-card.scss:18]: O, I didn't know that. Will revert the aspect ratio changes then.

File: sass/components/_icon.scss [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
-        }
-    }
+  &--#{nth($icon, 1)} {
+    height: var(--icon-height);
+    width: calc(var(--icon-height) * #{$ratio});
+    -webkit-mask-image: url('/assets/icon-#{nth($icon, 1)}.svg');
+    mask-image: url('/assets/icon-#{nth($icon, 1)}.svg');
+  }
 }
+}
```
[the remark this answers is not in the document]
Comment (TrialDragon) [reply sass/components/_icon.scss:38]: There is an open PR for that #1013

File: sass/components/_image_compare.scss [hunk 3]
```diff
@@ -17,8 +19,8 @@ div.image-compare {
   outline-offset: -1px;
 
   // calculated
-  --clip-L-geo: inset(0 calc(100% - var(--slider-value) + var(--gap) / 2) 0 0);
-  --clip-R-geo: inset(0 0 0 calc(var(--slider-value) + var(--gap) / 2));
+  --clip-L-geo: inset(0 calc(100% - var(--slider-value) + math.div(var(--gap), 2)) 0 0);
+  --clip-R-geo: inset(0 0 0 calc(var(--slider-value) + math.div(var(--gap), 2)));
```
[the remark this answers is not in the document]
Comment (TrialDragon) [reply sass/components/_image_compare.scss:23]: Ope, yep. Image compare looks broken. Will see if removing that fixes it.

[the remark this answers is not in the document]
Comment (doup) [reply sass/components/_image_compare.scss:23]: OK, I can confirm this breaks the image comparison thingy:

image

File: sass/components/_image_compare.scss [hunk 4]
```diff
@@ -1,3 +1,5 @@
+@use "sass:math";
```
Comment (doup) [inline sass/components/_image_compare.scss:1]: Not needed anymore.

**Review by doup — APPROVED** [review]

File: sass/components/_asset-card.scss [hunk 5]
```diff
@@ -14,7 +15,7 @@ $asset-card-padding: 0.4rem;
     grid-area: banner;
     display: block;
     position: relative;
-    aspect-ratio: 16/9;
+    aspect-ratio: 16 / 9;
```
Comment (BD103) [inline sass/components/_asset-card.scss:18]: Is this meant to be a list or division?

Comment (TrialDragon) [reply sass/components/_asset-card.scss:18]: > **Nit**: `math.div` is not needed in this case, CSS accepts ratios in `x / y` form for `aspect-ratio`: https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio
> 
> I mean, you could use `math.div` as the property also accepts decimal numbers… but (for me at least) it just adds noise.

What Doup said above; so more like a list than division I think.

Comment (BD103) [reply sass/components/_asset-card.scss:18]: Whoops, didn't see that above. Makes sense, though.

Comment (doup) [reply sass/components/_asset-card.scss:18]: If I understand correctly, is similar to saying `16:9`, but [CSS choose](https://drafts.csswg.org/css-values-4/#ratios) `/` to separate width/height (?).

You can also write it as `1.7777` which CSS would interpret as `1.7777 / 1` (?). The only thing… that's not as useful/familiar as saying `16:9`.

**Comment by alice-i-cecile** [comment]
I'm going to wait on cleanup, then do a final test and merge :)

**Comment by BD103** [comment]
Could you update the README to mention that we're building with Zola 0.18?

https://github.com/bevyengine/bevy-website/blob/bbc525d0deb1dc2f99dc7fbba207f681ccb79b33/README.md?plain=1#L13

**Review by BD103 — APPROVED** [review]
I tested this locally and it looked good! Approved :)

**Review by RomainMazB — APPROVED** [review]
LGTM too :)

**Pull request merged**